pkg/provisioner/controller.go

```diff
@@ -638,3 +649,49 @@ func (c *obcController) objectBucketForClaimKey(key string) (*v1alpha1.ObjectBuc
 	}
 	return ob, nil
 }
+
+func (c *obcController) handleUpdateClaim(key string, obc *v1alpha1.ObjectBucketClaim, class *storagev1.StorageClass) error {
+	ob, err := getObjectBucketFromClaimKey(key, obc, c.libClientset, defaultRetryBaseInterval, defaultRetryTimeout)
+	if err != nil {
+		log.Error(err, "getting ob failed", "key", key)
+		return err
+	}
+	if reflect.DeepEqual(ob.Spec.Endpoint.AdditionalConfigData, obc.Spec.AdditionalConfig) {
+		return nil
+	} else {
+		logD.Info("Additional config changed, hence updating OB")
+		tmp := make(map[string]string)
+		var (
+			key   string
+			value string
+		)
+
+		for key, value = range ob.Spec.Endpoint.AdditionalConfigData {
+			tmp[key] = value
+			delete(ob.Spec.Endpoint.AdditionalConfigData, key)
+		}
+
+		for key, value = range obc.Spec.AdditionalConfig {
+			ob.Spec.Endpoint.AdditionalConfigData[key] = value
+		}
+
+		err = c.provisioner.Update(ob)
```

Comment:
I spent some time a while ago making sure that the `Create` method was idempotent.

What about the option of not creating a new `Update` method that has to be implemented by users (i.e., Rook) and instead call Create again? Or does that also pose different issues?